File: ex02/decomponhaeconverta.c
#include <stdio.h>

int main() {
    unsigned long long T;
    char S;

    if (scanf("%llu %c", &T, &S) != 2) {
        unsigned long long micro_por_segundo = 1000000;
        unsigned long long segundos_por_hora = 3600;
        unsigned long long horas_por_dia = 24;
        unsigned long long dias_por_semana = 7;

        unsigned long long segundos_totais = T / micro_por_segundo;
        unsigned long long horas_totais = segundos_totais / segundos_por_hora;

        unsigned long long semanas = horas_totais / (horas_por_dia * dias_por_semana);
        unsigned long long resto_horas_semanas = horas_totais % (horas_por_dia * dias_por_semana);

        unsigned long long dias = resto_horas_semanas / horas_por_dia;
        unsigned long long horas = resto_horas_semanas % horas_por_dia;

        printf("Decomposição do setor %c\n", S);
        printf("%llu Semana(s)\n", semanas);
        printf("%llu Dia(s)\n", dias);
        printf("%llu Hora(s)\n", horas);

        return 0;

    }
}

File: ex02/decomponhaeconverta.rs
use std::io;

fn main() {
    
    let mut input = String::new();

    io::stdin().read_line(&mut input).unwrap();

    let palavras: Vec<&str> = input.split_whitespace().collect();

    if palavras.len() == 2 {
        let t: u64 = palavras[0].parse().unwrap();
        let s: char = palavras[1].chars().next().unwrap();

        let micro_por_segundo: u64 = 1_000_000;
        let segundos_por_hora: u64 = 3600;
        let horas_por_dia: u64 = 24;
        let dias_por_semana: u64 = 7;

        let segundos_totais: u64 = t / micro_por_segundo;
        let horas_totais: u64 = segundos_totais / segundos_por_hora;

        let semanas: u64 = horas_totais / (horas_por_dia * dias_por_semana);
        let resto_horas_semanas: u64 = horas_totais % (horas_por_dia * dias_por_semana);

        let dias: u64 = resto_horas_semanas / horas_por_dia;
        let horas: u64 = resto_horas_semanas % horas_por_dia;

        println!("Decomposição do setor {}", s);
        println!("{} Semana(s)", semanas);
        println!("{} Dia(s)", dias);
        println!("{} Hora(s)", horas);
    }

}